controllers/tf_controller_reconcile.go

```diff
@@ -283,8 +283,10 @@ func (r *TerraformReconciler) reconcile(ctx context.Context, runnerClient runner
 	if readyCondition != nil && readyCondition.Status == metav1.ConditionUnknown {
 		log.Info("Last known action was: " + lastKnownAction)
 		cond := terraform.Status.Conditions[readyConditionIndex]
-		if cond.Reason == infrav1.PlannedWithChangesReason && strings.HasPrefix(cond.Message, "Plan generated") {
-			// do nothing
+		if cond.Reason == infrav1.PlannedWithChangesReason && strings.HasPrefix(cond.Message, "Plan generated") && terraform.Spec.PlanOnly {
+			if terraform.Spec.PlanOnly {
```

Comment:
Is this already known to be `true` from the condition immediately above?

Reply:
Good catch, the `&& terraform.Spec.PlanOnly` shouldn't be there.